### PR TITLE
Fix click listener kapt bug 

### PR DESCRIPTION
**Pull request by elihart**
This bug incorrectly generates click listener code.

Also cleans up how publishing is done

### CODE REVIEW

File: publishing.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Sets up publishing of release artifacts.
+// Note: Keep this script in sync across all airbnb open source projects.
+apply plugin: 'com.vanniktech.maven.publish'
+
+
+// https://vanniktech.github.io/gradle-maven-publish-plugin/other/
+publishing {
```
Comment (vinaygaba) [inline publishing.gradle:7]: Thank you for this file, I ended up using this for Showkase publishing as well 🙏

File: epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt [hunk 2]
```diff
@@ -43,9 +43,11 @@ class Type(val xType: XType, memoizer: Memoizer) {
                     xType.isTypeOf(CharSequence::class) || xType.isTypeOf(String::class) -> StringOrCharSequence
                     xType.typeName == ClassNames.EPOXY_STRING_ATTRIBUTE_DATA -> StringAttributeData
                     // We don't care about nullability for the purposes of type checking
-                    nonNullType == memoizer.viewOnClickListenerType -> ViewClickListener
-                    nonNullType == memoizer.viewOnLongClickListenerType -> ViewLongClickListener
-                    nonNullType == memoizer.viewOnCheckChangedType -> ViewCheckedChangeListener
+                    // Note, == does not work for type comparisons when comparing types between classpath
```
Comment (vinaygaba) [inline epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt:46]: any chance you could maybe describe the bug a bit more in the PR description. Just helps me understand a bit better. I can follow that the equals check wasn't what was needed but didn't understand what the difference in the generated code looked like

Comment (vinaygaba) [reply epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt:46]: is there any test case that could be added?

**Review by vinaygaba — APPROVED** [review]

File: epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt (hunk 2 above)
Comment (elihart) [reply epoxy-processor/src/main/java/com/airbnb/epoxy/processor/Type.kt:46]: hard to test because this happens when they are in different modules

**Pull request merged**